test/testsymboldatabase.cpp

```diff
@@ -1281,6 +1282,167 @@ class TestSymbolDatabase: public TestFixture {
         ASSERT(false == v.isReference());
     }
 
+    /*
+    * Verify that isVariableDeclaration can correctly differentiate between a function call and a variable declaration
+    * or use the user provided default determination when differentiation is not possible.
+    * See https://trac.cppcheck.net/ticket/9960
+    */
+    void isVariableDeclarationVsCall() {
+
+        //Note that isVariableDeclaration is called fairly early in the process of interpreting the code
+        //so to make sure it can't take advantage of information gained later in the process we don't
+        //test it directly here but rather test the results it generated.
+
+        //Test a simple case where foo has been declared as a function so it's definitely a function call
+        {
+            givenACodeSampleToTokenize var("int* a[5] = { new int, new int, new int, new int, new int };\n"
+                "void foo(auto x){}\n"
+                "int main() {\n"
+                "foo(a);\n" //From context this is definitely a function call
+                "}\n");
+            const Function* f = Token::findmatch(var.tokens(), "foo")->function();
+            const Variable* v = Token::findmatch(var.tokens(), "a")->variable();
+            for (auto t : var.tokens()->toend())
+            {
+                if (t->str() == "foo") { //foo is always a function
+                    ASSERT(f == t->function());
+                }
+
+                if (t->str() == "a") { //a is a variable
+                    ASSERT(v == t->variable());
+                }
+                else if (t->str() != "x") { //x is too but everything else is not
+                    ASSERT(nullptr == t->variable());
+                }
+            }
+        }
+
+        //Test a more complicated case where foo has been declared as a function so it's definitely a function call
+        {
+            givenACodeSampleToTokenize var("int* a[5] = { new int, new int, new int, new int, new int };\n"
+                "void foo(int x){}\n"
+                "int main() {\n"
+                "foo(*a[4]);\n" //From context this is definitely a function call
+                "}\n");
+            const Function* f = Token::findmatch(var.tokens(), "foo")->function();
+            const Variable* v = Token::findmatch(var.tokens(), "a")->variable();
+            for (auto t : var.tokens()->toend())
+            {
+                if (t->str() == "foo") { //foo is always a function
+                    ASSERT(f == t->function());
+                }
+
+                if (t->str() == "a") { //a is a variable
+                    ASSERT(v == t->variable());
+                }
+                else if (t->str() != "x") { //x is too but everything else is not
+                    ASSERT(nullptr == t->variable());
+                }
+            }
+        }
+
+        //Test a case where foo has been declared as a function in an awkward scope so it's definitely a function call
```

Comment:
This test case fails because the tokenizer generates the token "foo<4>" as a single token which doesn't match the existing function name.

Reply:
@danmar can you comment on this issue? Specifically should ```foo<4>``` ever be a single token or is this a bug in the tokenizer?